imagetest/testworkflow.go

```diff
@@ -565,3 +565,20 @@ func (t *TestWorkflow) AddSSHKey(user string) (string, error) {
 
 	return string(publicKey), nil
 }
+
+// CreateFirewallRule create firewall rule.
+func (t *TestWorkflow) CreateFirewallRule(firewallName, networkName, protocal string, ports []string) error {
```

Comment:
The naming and docs are unclear about what happens with "protocol" and "ports": Are they blocked? Allowed?

To make this clearer, I'd suggest changing the argument to a single "compute.Firewall" instance, so that it's clear what rule is being created.

Reply:
we intentionally do not take compute or daisy objects in this framework, that's not set in stone but if we want to change it i think it should be done conscientiously and not just for this test

Reply:
We could consider provide a Firewall struct defined by us just like Network